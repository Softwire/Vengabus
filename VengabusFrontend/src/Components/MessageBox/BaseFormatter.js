export class BaseFormatter {
    startsAndEndsWith = (inputString, startCharacter, endCharacter) => {
        let noInitialWhitespace = inputString.replace(/^\s*/g, "");
        let noWrappingWhitespace = noInitialWhitespace.replace(/\s*$/g, "");

        return this.startsAndEndsWith_NotIgnoringWhitespace(noWrappingWhitespace, startCharacter, endCharacter);
    }

    startsAndEndsWith_NotIgnoringWhitespace = (inputString, startCharacter, endCharacter) => {
        if (!inputString) {
            return false;
        }
        if (inputString[0] === startCharacter && inputString[inputString.length - 1] === endCharacter) {
            return true;
        }
        return false;
    }

    isMessageTooLong = (originalText) => {
        return originalText && originalText.length > 100000;
    }

    removeBlankLines = (text) => {
        if (!text) {
            return text;
        }
        return text.replace(/^\s*\n/gm, "");
    };

    matchWithoutWhitespaceAndQuotes = (text1, text2) => {
        //ignore quote marks as the JSON formatter sometimes removes these
        return text1.replace(/\s|'|"|`/g, "") === text2.replace(/\s|'|"|`/g, "");
    }

    getConfidenceMatch = (formattedObj, mightBeThisFormat) => {
        //return a float between 0 and 1 indicating how confident we are that the message is of the given format
        if (mightBeThisFormat && !formattedObj.errorMessage) {
            return formattedObj.warningMessage ? 0.5 : 0.9;
        }
        return 0;
    }

    formatResult = (originalText, title, formattingFunction, mightBeThisFormat) => {
        let formattedObj = { formatType: title };
        formattedObj.isMessageTooLongToFormat = this.isMessageTooLong(originalText);
        if (formattedObj.isMessageTooLongToFormat) {
            formattedObj.errorMessage = 'Long message: only messages under 100,000 characters in length are formatted.';
        } else {
            try {
                formattedObj.formattedText = formattingFunction(originalText);
                if (formattedObj.formattedText && !this.matchWithoutWhitespaceAndQuotes(formattedObj.formattedText, originalText)) {
                    formattedObj.warningMessage = `The ${title} formatter changed the text of this data. This was probably just to 'heal' malformed ${title}, but we can't be certain.`;
                }
            }
            catch (error) {
                formattedObj.errorMessage = error.toString();
            }
        }
        formattedObj.matchConfidence = this.getConfidenceMatch(formattedObj, mightBeThisFormat);
        return formattedObj;
    }
}