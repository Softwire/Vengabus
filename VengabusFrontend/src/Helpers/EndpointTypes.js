var EndpointTypes = Object.freeze({
    QUEUE: "queue",
    TOPIC: "topic",
    SUBSCRIPTION: "subscription",
    MESSAGE: "message",
    DEADLETTER: "deadletters"
});


var typeToTitle = (type) => {
    switch (type) {
        case EndpointTypes.MESSAGE:
            return "Messages";
        case EndpointTypes.SUBSCRIPTION:
            return "Subscriptions";
        case EndpointTypes.QUEUE:
            return "Queues";
        case EndpointTypes.TOPIC:
            return "Topics";
        case EndpointTypes.DEADLETTER:
            return "Deadletters";
        default:
            throw new Error('Invalid endpoint type.');
    }
}
export { EndpointTypes, typeToTitle };