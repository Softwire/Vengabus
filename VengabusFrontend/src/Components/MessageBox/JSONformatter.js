import { BaseFormatter } from './BaseFormatter';

export class JSONformatter {
    constructor() {
        this.BaseFormatter = new BaseFormatter();
    }

    mightBeJSON = (originalText) => {
        return this.BaseFormatter.startsAndEndsWith(originalText, '{', '}') || this.BaseFormatter.startsAndEndsWith(originalText, '[', ']');
    }

    formatJSONtext = (originalText) => {
        const propertyFilter = null; //include all properties of the JSON string in the output
        const indentSize = 4;
        return this.BaseFormatter.removeBlankLines(JSON.stringify(JSON.parse(originalText), propertyFilter, indentSize));
    }

    getFormatResult = (originalText) => {
        let formattedObj = this.BaseFormatter.formatResult(originalText, 'JSON', this.formatJSONtext, this.mightBeJSON(originalText));
        return formattedObj;
    }
}