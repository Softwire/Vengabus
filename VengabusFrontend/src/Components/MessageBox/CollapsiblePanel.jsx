import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';
import { sharedSizesAndDimensions } from '../../Helpers/SharedSizesAndDimensions';

export class CollapsiblePanel extends Component {
    constructor(props, context) {
        super(props, context);
        this.state = {
            panellsOpen: this.props.isDefaultExpanded
        };
    }

    togglePanel = () => {
        this.setState({
            panellsOpen: !this.state.panellsOpen
        });
    }

    render() {
        const collapsiblePanelStyle = css`
            pre {
                background-color: white;
              
            }
            pre p {
                margin: 0;
            }
            
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const glyphiconStyle = css` /*we need the margins below to align the plus and minus glyphicons*/
            &.glyphicon-plus {
                margin-left: 1px;
                margin-right: calc(${sharedSizesAndDimensions.GLYPHICON_HORIZONTAL_MARGINS}px - 1px);
            }
            &.glyphicon-minus {
                margin-right: ${sharedSizesAndDimensions.GLYPHICON_HORIZONTAL_MARGINS}px;
            }
        `;
        return (
            <Panel defaultExpanded={this.props.isDefaultExpanded} className={collapsiblePanelStyle} onToggle={() => this.togglePanel()} >
                <Panel.Toggle>
                    <Panel.Heading>
                        <Panel.Title>
                            <Glyphicon glyph={this.state.panellsOpen ? 'minus' : 'plus'} className={glyphiconStyle} /><h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        </Panel.Title>
                    </Panel.Heading>
                </Panel.Toggle>
                <Panel.Collapse>
                    <Panel.Body>{this.props.children}</Panel.Body>
                </Panel.Collapse>
            </Panel >

        );
    }

}