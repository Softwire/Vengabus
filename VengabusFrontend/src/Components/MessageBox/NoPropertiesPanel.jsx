import React, { Component } from 'react';
import { Panel, Glyphicon } from 'react-bootstrap';
import { css } from 'emotion';
import { sharedSizesAndDimensions } from '../../Helpers/SharedSizesAndDimensions';
import { panelMediumGrey } from '../../colourScheme';

export class NoPropertiesPanel extends Component {

    render() {
        const noPropertiesPanelStyle = css`
            .panel-heading pre {
                float: right;
                margin-right: 10px;
                padding: 10px; /*to align the no properties text with the h4 title*/
                background: transparent;
            }
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const glyphiconStyle = css` /*we need the margins below to align with the minus glyphicon in CollapsiblePanel*/
            &.glyphicon-plus {
                margin-left: 1px;
                margin-right: calc(${sharedSizesAndDimensions.GLYPHICON_HORIZONTAL_MARGINS}px - 1px);
                color: ${panelMediumGrey};
            }
        `;
        return (
            <Panel className={noPropertiesPanelStyle} >
                <Panel.Heading>
                    <Panel.Title >
                        <Glyphicon glyph={'plus'} className={glyphiconStyle} />
                        <h4 className={headerStyle}>{this.props.panelTitle}</h4>
                        <pre>{`There are no properties to display`}</pre>
                    </Panel.Title>
                </Panel.Heading>
            </Panel >

        );
    }

}