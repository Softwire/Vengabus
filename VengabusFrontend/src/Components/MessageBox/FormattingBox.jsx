import React, { Component } from 'react';
import { css } from 'react-emotion';
import { Alert, Tabs, Tab } from 'react-bootstrap';
import { JSONformatter } from './JSONformatter';
import { XMLformatter } from './XMLformatter';
import { OriginalFormatter } from './OriginalFormatter';
import _ from 'lodash';
import { sharedSizesAndDimensions } from '../../Helpers/SharedSizesAndDimensions';
import { Glyphicon } from 'react-bootstrap';
import { reactBoostrapDangerRedText, reactBoostrapDangerRedBackground, reactBoostrapWarningYellowText, reactBoostrapWarningYellowBackground } from '../../colourScheme';

export class FormattingBox extends Component {
    constructor(props) {
        super(props);
        this.JSONformatter = new JSONformatter();
        this.XMLformatter = new XMLformatter();
        this.OriginalFormatter = new OriginalFormatter();
    }

    formatterErrorAlert(message, key) {
        return this.alertBlock("danger", key, `The formatter threw an error whilst trying to format the text of this data:`, message);
    }

    formatterWarningAlert(message, key) {
        return this.alertBlock(
            "warning",
            key,
            `The formatter returned a warning whilst trying to format the text of this data:`,
            message,
            'See "Original Text" for the unformatted data.'
        );
    }

    noTextAlert(key) {
        return this.alertBlock("danger", key, "The formatter didn't return any text to display.");
    }

    alertBlock(alertType, key, ...messageLines) {
        const lines = messageLines.map((line, index) =>
            <p key={index}>{line}</p>
        );
        return (
            <Alert bsStyle={alertType} key={key} id={key}>
                {lines}
            </Alert >
        );
    }

    chooseDefaultFormat(formattedObjectArray) {
        //choose which tab in messageBox to default to
        const mostLikelyFormat = _.maxBy(formattedObjectArray, 'matchConfidence');
        return formattedObjectArray.indexOf(mostLikelyFormat);
    }

    getContentToDisplay(formattingAttemptResult) {
        let contentToDisplay = [];
        const error = formattingAttemptResult.errorMessage;
        const warning = formattingAttemptResult.warningMessage;
        const formattedText = formattingAttemptResult.formattedText;
        if (error) {
            contentToDisplay.push(this.formatterErrorAlert(error, formattingAttemptResult.formatType + "error"));
        }
        if (warning) {
            contentToDisplay.push(this.formatterWarningAlert(warning, formattingAttemptResult.formatType + "warning"));
        }
        if (formattedText) {
            const textStyle = formattingAttemptResult.formatType === 'Original' ? css`white-space: pre-wrap;` : ''; //wordwrap original text
            contentToDisplay.push(<pre key={formattingAttemptResult.formatType} className={textStyle} id={formattingAttemptResult.formatType}>{formattedText}</pre>);
        }
        if (contentToDisplay.length === 0) {
            formattingAttemptResult.errorMessage = "No text returned from the formatter"; //needed so that getFormatterTab recongises that this is indeed an error
            contentToDisplay.push(this.noTextAlert(formattingAttemptResult.formatType + "warning"));
        }
        return contentToDisplay;
    }

    getFormatterTab(formattingAttemptResult, tabKey) {
        const contentToDisplay = this.getContentToDisplay(formattingAttemptResult);
        let tabTitle = [formattingAttemptResult.formatType];
        let errorState;
        const glyphStyle = css`
            margin-left: 5px;
        `;
        if (formattingAttemptResult.errorMessage) {
            tabTitle.push(<Glyphicon glyph="exclamation-sign" key="errorGlyph" className={glyphStyle} />);
            errorState = "formatterError";
        } else if (formattingAttemptResult.warningMessage) {
            tabTitle.push(<Glyphicon glyph="warning-sign" key="warningGlyph" className={glyphStyle} />);
            errorState = "formatterWarning";
        }
        return <Tab title={tabTitle} eventKey={tabKey} key={tabKey} tabClassName={errorState}>{contentToDisplay}</ Tab>;
    }

    render() {
        const originalText = this.props.message;
        const formatters = [this.OriginalFormatter, this.XMLformatter, this.JSONformatter];
        const formattedObjects = formatters.map(formatter => formatter.getFormatResult(originalText));
        const messageTabsArray = formattedObjects.map((formattedObj, index) => this.getFormatterTab(formattedObj, index));
        const defaultTabToDisplay = this.chooseDefaultFormat(formattedObjects);

        const tabStyle = css`
            .nav-tabs {
                margin-bottom: 15px;
            }
            .tab-content {
                overflow: auto;
                max-height: calc(${sharedSizesAndDimensions.MESSAGEBOX_MODAL_HEIGHT}vh - 326px); /*326px is approximately the height of the MessageBox modal without FormattingBox*/

                pre {
                    margin-bottom: 0;
                }
            }
            .formatterError {
                a { /*colour the tab when there's an error*/
                    color: ${reactBoostrapDangerRedText};
                    background: ${reactBoostrapDangerRedBackground};
                }
                &.active>a {
                    &, &:focus, &:hover { /*need all these cases to override the default style*/
                        color: ${reactBoostrapDangerRedText};
                    }
                }
            }
            .formatterWarning {
                a { /*colour the tab when there's a warning*/
                    color: ${reactBoostrapWarningYellowText};
                    background: ${reactBoostrapWarningYellowBackground};
                }
                &.active>a {
                    &, &:focus, &:hover { /*need all these cases to override the default style*/
                        color: ${reactBoostrapWarningYellowText};
                    }
                }
            }
        `;



        return (
            <div>
                <Tabs animation={false} defaultActiveKey={defaultTabToDisplay} id="message-formatting-tabs" className={tabStyle}>
                    {messageTabsArray}
                </Tabs>
            </div>
        );
    }
}
