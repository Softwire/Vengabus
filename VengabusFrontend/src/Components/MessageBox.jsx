import React, { Component } from 'react';
import { Modal, Button } from 'react-bootstrap';
import { FormattingBox } from './FormattingBox';
import { CopyTextButton } from './CopyTextButton';
import { css } from 'emotion';
import { CollapsiblePanel } from './CollapsiblePanel';

export class MessageBox extends Component {

    convertMessagePropertiesToJSXArray = (propertiesArray) => {
        const propertiesJSX = [];
        for (let propName in propertiesArray) {
            let currentPropValue = propertiesArray[propName];
            if (propName !== 'messageId') {
                propertiesJSX.push(
                    <p key={propName}>{propName + ': ' + String(currentPropValue)}</p> /*string conversion needed to display booleans properly*/
                );
            }
        }
        return propertiesJSX.length > 0 ? propertiesJSX : null;
    }

    render() {
        const buttonStyle = css`
            float: right;
            margin-left: 5px;
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const messageIdStyle = css`
            display: inline-block;
        `;
        const panelStyle = css`
            .panel {
                margin-bottom: 15px;
            }
        `;
        const message = this.props.message;
        if (!message) {
            return null;
        }
        const preDefinedPropsJSX = this.convertMessagePropertiesToJSXArray(message.predefinedProperties) || <p>There are no pre-defined properties to display</p>;
        const customPropsJSX = this.convertMessagePropertiesToJSXArray(message.customProperties) || <p>There are no user-defined properties to display</p>;
        return (
            <div className="static-modal" >
                <Modal show={this.props.show} onHide={this.props.handleClose} >
                    <Modal.Header>
                        <Modal.Title className={headerStyle}>Message Id:&nbsp;</Modal.Title>
                        <Modal.Title className={messageIdStyle}>{message.predefinedProperties.messageId}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body className={panelStyle}>
                        <CollapsiblePanel panelTitle={"Pre-defined Properties"}>
                            <pre>{preDefinedPropsJSX}</pre>
                        </CollapsiblePanel>
                        <CollapsiblePanel panelTitle={"User-defined Properties"}>
                            <pre>{customPropsJSX}</pre>
                        </CollapsiblePanel>
                        <FormattingBox
                            data={message.messageBody}
                        />
                    </Modal.Body>

                    <Modal.Footer>
                        <Button className={buttonStyle} onClick={this.props.handleClose}>Close</Button>
                        <CopyTextButton className={buttonStyle} text={this.props.messageBody} />
                    </Modal.Footer>
                </Modal>
            </div>

        );
    }
}