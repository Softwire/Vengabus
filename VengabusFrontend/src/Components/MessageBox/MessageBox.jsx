import React, { Component } from 'react';
import { Modal, Button, ButtonToolbar } from 'react-bootstrap';
import { CopyTextButton } from '../CopyTextButton';
import { css } from 'emotion';
import { CollapsiblePanel } from '../CollapsiblePanel';
import { PAGES, pageSwitcher } from '../../Pages/PageSwitcherService';
import { FormattingBox } from './FormattingBox';

export class MessageBox extends Component {

    convertMessagePropertiesToJSXArray = (propertiesArray) => {
        const propertiesJSX = [];
        for (let propName in propertiesArray) {
            let currentPropValue = propertiesArray[propName];
            if (propName !== 'messageId') {
                propertiesJSX.push(
                    <p key={propName}>{propName + ': ' + String(currentPropValue)}</p> /*string conversion needed to display booleans properly*/
                );
            }
        }
        return propertiesJSX.length > 0 ? propertiesJSX : null;
    }

    handleReplayMessage = (message) => {
        //qq change hardcoded recipientIsQueue and selectedQueue later
        pageSwitcher.switchToPage(
            PAGES.SendMessagePage,
            { message: message, recipientIsQueue: true, selectedQueue: 'demoqueue1' }
        );
    }

    render() {
        const buttonToolbarStyle = css`
            margin-right: -8px; /* This counteracts the right margin for the "first" (right-most) button. See comment below. */
            
            &.btn-toolbar .btn {
                margin-right: 8px; /* The tooltip library's layout code doesn't handle margins, so we have to have the margin on the right. */
                margin-left: 0px;
                float: right;
            }
        `;
        const headerStyle = css`
            font-weight: bold;
            display: inline-block;
        `;
        const messageIdStyle = css`
            display: inline-block;
        `;
        const panelStyle = css`
            .panel {
                margin-bottom: 15px;
            }
        `;
        const message = this.props.message;
        if (!message) {
            return null;
        }
        const preDefinedPropsJSX = this.convertMessagePropertiesToJSXArray(message.predefinedProperties) || <p>There are no pre-defined properties to display</p>;
        const customPropsJSX = this.convertMessagePropertiesToJSXArray(message.customProperties) || <p>There are no user-defined properties to display</p>;
        return (
            <div className="static-modal" >
                <Modal show={this.props.show} onHide={this.props.handleClose} id="messageBoxModal" >
                    <Modal.Header>
                        <Modal.Title className={headerStyle}>Message Id:&nbsp;</Modal.Title>
                        <Modal.Title className={messageIdStyle}>{message.predefinedProperties.messageId}</Modal.Title>
                    </Modal.Header>

                    <Modal.Body className={panelStyle} id="messageBoxModalBody">
                        <CollapsiblePanel panelTitle={"Pre-defined Properties"}>
                            <pre>{preDefinedPropsJSX}</pre>
                        </CollapsiblePanel>
                        <CollapsiblePanel panelTitle={"User-defined Properties"}>
                            <pre>{customPropsJSX}</pre>
                        </CollapsiblePanel>
                        <FormattingBox message={this.props.message.messageBody} />
                    </Modal.Body>

                    <Modal.Footer>
                        <ButtonToolbar className={buttonToolbarStyle}>
                            { /*Note that these buttons are rendered in order, Right-to-Left*/}
                            <Button onClick={this.props.handleClose} id="messageBoxClose">Close</Button>
                            <CopyTextButton text={message.messageBody} id="messageBoxCopy" />
                            <Button onClick={() => this.handleReplayMessage(message)} id="messageBoxReplayMessage">Replay Message to demoqueue1</ Button>
                        </ButtonToolbar>
                    </Modal.Footer>
                </Modal>
            </div>

        );
    }
}