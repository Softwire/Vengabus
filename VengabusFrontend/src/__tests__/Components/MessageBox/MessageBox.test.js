import { MessageBox } from '../../../Components/MessageBox/MessageBox';
import React from 'react';
import { shallow, mount } from 'enzyme';
import { EndpointTypes } from '../../../Helpers/EndpointTypes';
import toJson from 'enzyme-to-json';

describe('MessageBox', () => {
    const testMessage = {
        uniqueId: "1a",
        customProperties: { customProp1: "forty-two", customProp2: 42 },
        messageBody: "Hello world!",
        predefinedProperties: { forcePersistence: false, messageId: "test ID", scheduledEnqueueTimeUtc: "0001-01-01T00:00:00Z", sequenceNumber: 45317471370415620, size: 173, state: "Active", timeToLive: "10675199.02:48:05.4775807" }
    };

    const arePanelsCollapsed = (wrapper) => {
        const panels = wrapper.find('#messageBoxModalBody .panel');
        const numberOfCollapsedPanels = panels.find('a.collapsed').length;
        if (numberOfCollapsedPanels === 2 && panels.length === 2) {
            return true;
        }
        else if (numberOfCollapsedPanels === 0 && panels.length === 2) {
            return false;
        }
        throw new Error("Could not find the expected number of panels in MessageBox");
    };

    const countGlyphicons = (wrapper) => {
        const plusGlyphiconCount = wrapper.find("#messageBoxModalBody .panel .panel-heading .glyphicon-plus").length;
        const minusGlyphiconCount = wrapper.find("#messageBoxModalBody .panel .panel-heading .glyphicon-minus").length;
        return { plus: plusGlyphiconCount, minus: minusGlyphiconCount };
    }

    const clickPanelGlyphicons = (wrapper) => {
        const glyphicons = wrapper.find("#messageBoxModalBody .panel .panel-heading .glyphicon");
        glyphicons.forEach((glyph) => glyph.simulate("click"));
    }

    it('renders correctly with given props', () => {
        const wrapper = shallow(<MessageBox
            message={testMessage}
            show
            endpointType={EndpointTypes.QUEUE}
            messageType={EndpointTypes.MESSAGE}
            parentName={'testing'}
            endpointName={'test name'}
            closeParentModal={() => { }}
        />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('the properties panels toggle as expected', () => {
        const wrapper = mount(<MessageBox
            message={testMessage}
            show
            endpointType={EndpointTypes.QUEUE}
            messageType={EndpointTypes.MESSAGE}
            parentName={'testing'}
            endpointName={'test name'}
            closeParentModal={() => { }}
        />);
        //the properties panels should be closed by default
        expect(countGlyphicons(wrapper)).toEqual({ plus: 2, minus: 0 });
        expect(arePanelsCollapsed(wrapper)).toBe(true);

        //clicking on the glyphicons should expand the panels and toggle the glyphicons
        clickPanelGlyphicons(wrapper);
        wrapper.update();
        expect(countGlyphicons(wrapper)).toEqual({ plus: 0, minus: 2 });
        expect(arePanelsCollapsed(wrapper)).toBe(false);

        //clicking on the glyphicons again should collapse the panels and toggle the glyphicons
        clickPanelGlyphicons(wrapper);
        wrapper.update();
        expect(countGlyphicons(wrapper)).toEqual({ plus: 2, minus: 0 });
        expect(arePanelsCollapsed(wrapper)).toBe(true);
    });

});
