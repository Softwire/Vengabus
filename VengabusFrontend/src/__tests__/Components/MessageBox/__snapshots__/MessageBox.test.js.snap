// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageBox renders correctly with given props 1`] = `
.emotion-3 {
  margin-right: -8px;
}

.emotion-3.btn-toolbar .btn {
  margin-right: 8px;
  margin-left: 0px;
  float: right;
}

.emotion-0 {
  font-weight: bold;
  display: inline-block;
}

.emotion-1 {
  display: inline-block;
}

.emotion-2 .panel {
  margin-bottom: 15px;
}

.emotion-2 .panel .panel-heading {
  padding: 0 15px;
  color: #333;
  border-color: #ddd;
  overflow: auto;
  background-image: linear-gradient(to bottom,#f5f5f5 0,#e8e8e8 100%);
  background-repeat: repeat-x;
}

.emotion-2 .panel .panel-heading pre {
  white-space: pre-wrap;
}

.emotion-2 .panel .panel-body {
  padding: 0;
}

.emotion-2 .panel pre {
  border: none;
  margin: 0;
}

@media only screen and (min-width:750px) {
  .emotion-4 .modal-dialog {
    max-width: 750px;
    width: 100%;
  }
}

@media only screen and (max-width:750px) {
  .emotion-4 .modal-dialog {
    width: auto;
  }
}

.emotion-4 .modal-dialog {
  margin: 30px auto;
}

.emotion-4 .modal-content {
  overflow: auto;
  max-height: 92vh;
}

<div
  className="static-modal"
>
  <Modal
    animation={true}
    autoFocus={true}
    backdrop={true}
    bsClass="modal"
    className="emotion-4"
    dialogComponentClass={[Function]}
    enforceFocus={true}
    id="messageBoxModal"
    keyboard={true}
    manager={
      ModalManager {
        "add": [Function],
        "containers": Array [],
        "data": Array [],
        "handleContainerOverflow": true,
        "hideSiblingNodes": true,
        "isTopModal": [Function],
        "modals": Array [],
        "remove": [Function],
      }
    }
    onHide={[Function]}
    renderBackdrop={[Function]}
    restoreFocus={true}
    show={true}
  >
    <ModalHeader
      bsClass="modal-header"
      closeButton={false}
      closeLabel="Close"
    >
      <ModalTitle
        bsClass="modal-title"
        className="emotion-0"
        componentClass="h4"
      >
        Message Id:Â 
      </ModalTitle>
      <ModalTitle
        bsClass="modal-title"
        className="emotion-1"
        componentClass="h4"
      >
        test ID
      </ModalTitle>
    </ModalHeader>
    <ModalBody
      bsClass="modal-body"
      className="emotion-2"
      componentClass="div"
      id="messageBoxModalBody"
    >
      <CollapsiblePanel
        panelTitle="Pre-defined Properties"
      >
        <pre>
          <p
            key="forcePersistence"
          >
            forcePersistence: false
          </p>
          <p
            key="scheduledEnqueueTimeUtc"
          >
            scheduledEnqueueTimeUtc: 0001-01-01T00:00:00Z
          </p>
          <p
            key="sequenceNumber"
          >
            sequenceNumber: 45317471370415620
          </p>
          <p
            key="size"
          >
            size: 173
          </p>
          <p
            key="state"
          >
            state: Active
          </p>
          <p
            key="timeToLive"
          >
            timeToLive: 10675199.02:48:05.4775807
          </p>
        </pre>
      </CollapsiblePanel>
      <CollapsiblePanel
        panelTitle="User-defined Properties"
      >
        <pre>
          <p
            key="customProp1"
          >
            customProp1: forty-two
          </p>
          <p
            key="customProp2"
          >
            customProp2: 42
          </p>
        </pre>
      </CollapsiblePanel>
      <FormattingBox
        message="Hello world!"
      />
    </ModalBody>
    <ModalFooter
      bsClass="modal-footer"
      componentClass="div"
    >
      <ButtonToolbar
        bsClass="btn-toolbar"
        className="emotion-3"
      >
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          id="messageBoxClose"
        >
          Close
        </Button>
        <DeleteSingleMessageButton
          disabled={false}
          endpointName="test name"
          endpointType="queue"
          messageId="test ID"
          messageType="message"
          onDeletionEnd={[Function]}
          onDeletionStart={[Function]}
          uniqueId="1a"
        />
        <CopyTextButton
          disabled={false}
          id="messageBoxCopy"
          text="Hello world!"
        />
        <DownloadMessagesFileButton
          disabled={false}
          downloadButtonText="Download"
          fileName="message_test ID"
          getMessages={[Function]}
          id="messageBoxDownloadMessageButton"
        />
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          disabled={false}
          id="messageBoxReplayMessage"
          onClick={[Function]}
        >
          Replay Message to test name
        </Button>
      </ButtonToolbar>
    </ModalFooter>
  </Modal>
</div>
`;
